Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -3127,6 +3130,74 @@ extension DocumentationContext {
         }
         diagnosticEngine.emit(problems)
     }
+    
+    func analyzeAlternateDeclarations() {
+        var problems = [Problem]()
+        for node in topicGraph.nodes.values {
+            guard let entity = try? self.entity(with: node.reference) else { continue }
+            
+            var sourceLanguageToReference: [SourceLanguage: AlternateDeclaration] = [:]
+            for alternateDeclaration in entity.metadata?.alternateDeclarations ?? [] {
+                guard case .resolved(.success(let counterpartReference)) = alternateDeclaration.counterpart,
+                        let counterpartEntity = try? self.entity(with: counterpartReference) else {
+                    continue
```

Comment:
We will already emit a diagnostic here so I don't think it's also needed here, WDYT?
https://github.com/swiftlang/swift-docc/blob/bc6df5117b120f17d27faf51451a61215ab141fc/Sources/SwiftDocC/Infrastructure/DocumentationContext.swift#L614-L621

Since this ultimately ends up adding this diagnostic:
https://github.com/swiftlang/swift-docc/blob/bc6df5117b120f17d27faf51451a61215ab141fc/Sources/SwiftDocC/Semantics/ReferenceResolver.swift#L117

(I can add a comment to explain why we don't need a diagnostic here)